### PR TITLE
feat: added an option for prettier and a extended eslint config (#1425)

### PR DESCRIPTION
**Pull request by juliusmarminge**
cc @FinnDore

### CODE REVIEW

**Comment by FinnDore** [comment]
ah some of the e2e's. ill get on that

**Comment by FinnDore** [comment]
#1428  :)

File: cli/src/installers/dependencyVersionMap.ts [hunk 1]
```diff
@@ -26,5 +23,14 @@ export const dependencyVersionMap = {
...
   superjson: "1.12.2",
+
+  // Prettier
+  prettier: "^2.8.8",
+  "prettier-plugin-tailwindcss": "^0.2.8",
+  "@types/prettier": "^2.7.2",
+  "@ianvs/prettier-plugin-sort-imports": "^3.7.2",
+
+  // Eslint
+  "eslint-config-next": "^13.4.2",
```
Comment (juliusmarminge) [inline cli/src/installers/dependencyVersionMap.ts:34]: so this part doesn't make sense to me. why is `eslint-config-next` not in the default but a more opinionated option, strict ts linting, is?

File: cli/template/extras/config/_eslintrc.cjs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+      parserOptions: {
+        project: path.join(__dirname, "tsconfig.json"),
+      },
+    },
+  ],
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    project: path.join(__dirname, "tsconfig.json"),
+  },
+  plugins: ["@typescript-eslint"],
```
Comment (juliusmarminge) [inline cli/template/extras/config/_eslintrc.cjs:21]: do we wanna slap some react-hook lint rules in here?

also, eslint-config-prettier?

File: cli/template/extras/config/_prettier.config.cjs
Comment (juliusmarminge) [inline cli/template/extras/config/_prettier.config.cjs:1]: would it be better if we just build these files? very small diff to have them separate files + very easy to modify them typesafely?

File: cli/template/extras/config/prettier-with-tailwind.config.cjs
Comment (juliusmarminge) [inline cli/template/extras/config/prettier-with-tailwind.config.cjs:1]: ^^

File: www/.prettierrc.cjs [hunk 3]
```diff
@@ -5,6 +5,7 @@ const config = {
   plugins: [
     ...baseConfig.plugins,
     require.resolve("prettier-plugin-astro"),
+    require.resolve("@ianvs/prettier-plugin-sort-imports"),
```
Comment (juliusmarminge) [inline www/.prettierrc.cjs:8]: isn't this already in root?

**Review by juliusmarminge — COMMENTED** [review]
left a few comments @FinnDore

**Review by juliusmarminge — COMMENTED** [review]
left a few comments @FinnDore

File: cli/src/cli/index.ts [hunk 4]
```diff
@@ -329,3 +345,16 @@ const promptImportAlias = async (): Promise<string> => {
...
   return importAlias;
 };
+
+const promtPrettierAndExtendedEslint = async (): Promise<boolean> => {
+  const { prettierAndExtendedEslint } = await inquirer.prompt<
+    Pick<CliFlags, "prettierAndExtendedEslint">
+  >({
+    name: "prettierAndExtendedEslint",
+    type: "confirm",
+    message: "Would you like to use prettier and extended an eslint config?",
```
Comment (c-ehrlich) [inline cli/src/cli/index.ts:355]:
```suggestion
    message: "Would you like to use prettier and an extended eslint config?",
```

File: cli/src/installers/dependencyVersionMap.ts (hunk 1 above)
Comment (FinnDore) [reply cli/src/installers/dependencyVersionMap.ts:34]: Think i forgot it in both configs. Imo we should have everything that is in the more strict version at the moment in the base version, so i dont really know what we want in the more strict version, those hooks rules you mentioned maybe?

File: www/.prettierrc.cjs (hunk 3 above)
Comment (FinnDore) [reply www/.prettierrc.cjs:8]: ye, fixed

**Comment by FinnDore** [comment]
image
A bunch of my comments are pending. not sure what thats about.

**Comment by FinnDore** [comment]
Ive closed #1431 as i think we probably need to work out how this eslint config should look. Im very much for the addtion of optional prettier however im unsure what the extended eslint config should look like as i never really go deep on eslint configs. Julius has potentially mentioned some hook rules might be good for example. Welcome peoples thoughts here :)

**Pull request closed without merging**